File: p2pnet/android/app/src/main/java/com/example/p2pnet/data/local/LocalPrefs.kt
package com.example.p2pnet.data.local

import android.content.Context
import android.content.SharedPreferences
import com.example.p2pnet.util.Constants

class LocalPrefs(context: Context) {

    private val prefs: SharedPreferences = context.getSharedPreferences(
        Constants.PREFS_NAME, Context.MODE_PRIVATE
    )

    var serverHost: String
        get() = prefs.getString(Constants.KEY_SERVER_HOST, Constants.DEFAULT_SERVER_HOST) ?: Constants.DEFAULT_SERVER_HOST
        set(value) = prefs.edit().putString(Constants.KEY_SERVER_HOST, value).apply()

    var serverPort: Int
        get() = prefs.getInt(Constants.KEY_SERVER_PORT, Constants.DEFAULT_SERVER_PORT)
        set(value) = prefs.edit().putInt(Constants.KEY_SERVER_PORT, value).apply()

    var username: String?
        get() = prefs.getString(Constants.KEY_USERNAME, null)
        set(value) = prefs.edit().putString(Constants.KEY_USERNAME, value).apply()

    var loggedIn: Boolean
        get() = prefs.getBoolean(Constants.KEY_LOGGED_IN, false)
        set(value) = prefs.edit().putBoolean(Constants.KEY_LOGGED_IN, value).apply()

    fun clearSession() {
        prefs.edit()
            .remove(Constants.KEY_USERNAME)
            .putBoolean(Constants.KEY_LOGGED_IN, false)
            .apply()
    }
}

File: p2pnet/ios/p2pnet/Data/Local/LocalPrefs.swift
import Foundation

struct Constants {
    static let defaultServerHost = "deepstack.tech"
    static let defaultServerPort = 10000
    static let prefsName = "p2pnet_prefs"
    static let keyServerHost = "server_host"
    static let keyServerPort = "server_port"
    static let keyUsername = "username"
    static let keyLoggedIn = "logged_in"
}

class LocalPrefs {
    private let defaults: UserDefaults

    init() {
        defaults = UserDefaults(suiteName: Constants.prefsName) ?? .standard
    }

    var serverHost: String {
        get { defaults.string(forKey: Constants.keyServerHost) ?? Constants.defaultServerHost }
        set { defaults.set(newValue, forKey: Constants.keyServerHost) }
    }

    var serverPort: Int {
        get { defaults.integer(forKey: Constants.keyServerPort) != 0 ? defaults.integer(forKey: Constants.keyServerPort) : Constants.defaultServerPort }
        set { defaults.set(newValue, forKey: Constants.keyServerPort) }
    }

    var username: String? {
        get { defaults.string(forKey: Constants.keyUsername) }
        set { defaults.set(newValue, forKey: Constants.keyUsername) }
    }

    var loggedIn: Bool {
        get { defaults.bool(forKey: Constants.keyLoggedIn) }
        set { defaults.set(newValue, forKey: Constants.keyLoggedIn) }
    }

    func clearSession() {
        defaults.removeObject(forKey: Constants.keyUsername)
        defaults.set(false, forKey: Constants.keyLoggedIn)
    }
}